jbake-core/src/main/java/org/jbake/app/ContentStore.java

```diff
@@ -113,17 +122,23 @@ private void startupIfEnginesAreMissing() {
 
         // If an instance of Orient was previously shutdown all engines are removed.
         // We need to startup Orient again.
-        if ( Orient.instance().getEngines().size() == 0 ) {
+        if (Orient.instance().getEngines().isEmpty()) {
             Orient.instance().startup();
         }
         OLogManager.instance().setWarnEnabled(true);
     }
 
     public void drop() {
+        db.activateOnCurrentThread();
```

Comment:
Please change this to call private method `activateOnCurrentThread()` to have a single entry point.